Source/SPConnectionController.m

```diff
@@ -1326,6 +1364,35 @@ - (IBAction)exportFavorites:(id)sender
 	 }];
 }
 
+#pragma mark -
+#pragma mark Time Zone changes
+
+- (IBAction)didChangeSelectedTimeZone:(NSPopUpButton *)sender
+{
+	NSMenuItem *selectedItem = [sender selectedItem];
+	switch (selectedItem.tag) {
+		case SPUseServerTimeZoneTag:
+			[self setTimeZoneMode:SPConnectionTimeZoneModeUseServerTZ];
+			[self setTimeZoneIdentifier:@""];
+			break;
+		case SPUseSystemTimeZoneTag:
+			[self setTimeZoneMode:SPConnectionTimeZoneModeUseSystemTZ];
+			[self setTimeZoneIdentifier:@""];
+			break;
+		default:
+			[self setTimeZoneMode:SPConnectionTimeZoneModeUseFixedTZ];
+			[self setTimeZoneIdentifier:selectedItem.title];
+			break;
+	}
+
+
+	[standardTimeZoneField selectItemAtIndex:sender.indexOfSelectedItem];
+	[sshTimeZoneField selectItemAtIndex:sender.indexOfSelectedItem];
+	[socketTimeZoneField selectItemAtIndex:sender.indexOfSelectedItem];
```

Comment:
What's happening here? Why should the selectors change here?

Reply:
So, for most/all of the other fields, we use bindings set in the xib files to tie the standard/socket/ssh versions of a given entry field to the same property. That way, if the value of the socket username field is changed, for example, the standard and ssh username fields are updated automatically with the new value. Because I didn't go with bindings for timezone, I have to explicitly keep the fields in sync with this code. I could wrap each call in an if statement to only change the ones that are not the sender, but that felt overkill.